source/whats-new.txt

```diff
@@ -73,18 +74,40 @@ New versions of the {+driver-short+} can introduce the following types of breaki
      - Drops support for platform A, compiler B, or architecture C.
 
 .. _upcoming-breaking-changes:
+.. _version-4.0:
+
+What's New in 4.0
+-----------------
+
+.. warning:: Build System Breaking Changes
+
+   The v4.0 driver introduces the following breaking changes to the build system:
+   
+   - Drops support for the Boost and MNMLSTC/core external polyfill libraries. You can no
+     longer set the following configuration options:
+
+     - ``ENABLE_BSONCXX_POLY_USE_IMPLS``. This option is implicitly set to ``ON``.
+     - ``BSONCXX_POLY_USE_MNMLSTC``.
+     - ``BSONCXX_POLY_USE_SYSTEM_MNMLSTC``.
+     - ``BSONCXX_POLY_USE_BOOST``.
+
```

Comment:
```suggestion

.. warning:: API Breaking Changes

   The v4.0 driver introduces the following breaking changes to the API:

   - Drops support for configuring `bsoncxx::stdx::string_view` and `bsoncxx::stdx::optional<T>` as aliases for Boost or mnmlstc/core library equivalents. See "Build System Breaking Changes" above.

.. warning:: ABI Breaking Changes

   The v4.0 driver introduces the following breaking changes to the ABI:

   - Drops support for configuring `bsoncxx::stdx::string_view` and `bsoncxx::stdx::optional<T>` as aliases for Boost or mnmlstc/core library equivalents. See "Build System Breaking Changes" above.
```

Removal of external polyfill library support is also an API breaking change (_the_ important API breaking change for this release) and an ABI breaking change. Recommend adding a section/entry for each in addition to the build system breaking changes above (deliberately referenced under all three types of breaking changes).